### ISSUE TITLE
Would it help to add more detail to the names of Activation Layers

### ISSUE THREAD

**Issue by tubaybb321** [fchollet/keras#2873]
Currently, Layers are given default names that are determined only by their class name (with the exception of a uid that is appended). However, an Activation layer may represent a ReLU, SoftMax, Tanh etc.

I think that for diagnostic issues (or at least better identifying a net's architecture, particularly  if you use different activation layers), more information could be added to the default name. I'm proposing adding the following lines to the __init__ function of the Activation class, after the call to super:

```
        name_list = self.name.split('_')
        name_list.insert(-1,self.activation.func_name)
        self.name = '_'.join(name_list)
```

This way, one will see layer names like:

activation_relu_5

instead of just:

activation_5

I know this is an exceedingly minor issue, but it's a change I've made in my local copy and thought I might as well put it out there

**Comment by xingdi-eric-yuan** [comment]
minor change, in case of layer name contains underscore
```python
        name_list = self.name.rsplit('_', 1)
        name_list.insert(-1, self.activation.func_name)
        self.name = '_'.join(name_list)
```

**Comment by tubaybb321** [comment]
Is there a use case where there's a difference? I just tried this and didn't see one:

```
>>> z="purple_blue__red"
>>> z1=z.split('_')
>>> z2=z.rsplit('_',1)
>>> z1
['purple', 'blue', '', 'red']
>>> z2
['purple_blue_', 'red']
>>> z1.insert(-1,'HI')
>>> z2.insert(-1,'HI')
>>> '_'.join(z1)
'purple_blue__HI_red'
>>> '_'.join(z2)
'purple_blue__HI_red'
```

Am I missing something here? Is there a case where I'll get different results?

But, I think  rsplit('\_',1) makes it more clear I'm just interested in putting something before the final '_#'

**Comment by fchollet** [comment]
I think it would be inelegant to introduce special cases for naming based
on layer type. A single, consistent scheme is preferable. If you want to
name your layers differently, you can always pass a `name` argument to the
layers.
